pyuvsim/uvsim.py

```diff
@@ -276,101 +272,39 @@ def initialize_uvdata(uvtask_list, source_list_name, uvdata_file=None,
         history += (' Based on config files: ' + obs_param_file + ', '
                     + telescope_config_file + ', ' + antenna_location_file)
 
-    history += ' Npus = ' + str(mpi.get_Npus()) + '.'
-
-    task_freqs = []
-    task_bls = []
-    task_times = []
-    task_antnames = []
-    task_antnums = []
-    task_antpos = []
-    task_uvw = []
-    ant_1_array = []
-    ant_2_array = []
-    telescope_name = uvtask_list[0].telescope.name
-    telescope_location = uvtask_list[0].telescope.location.geocentric
-
-    source_0 = uvtask_list[0].source
-    freq_0 = uvtask_list[0].freq
-    for task in uvtask_list:
-        if not task.source == source_0:
-            continue
-        task_freqs.append(task.freq)
-
-        if task.freq == freq_0:
-            task_bls.append(task.baseline)
-            task_times.append(task.time)
-            task_antnames.append(task.baseline.antenna1.name)
-            task_antnames.append(task.baseline.antenna2.name)
-            ant_1_array.append(task.baseline.antenna1.number)
-            ant_2_array.append(task.baseline.antenna2.number)
-            task_antnums.append(task.baseline.antenna1.number)
-            task_antnums.append(task.baseline.antenna2.number)
-            task_antpos.append(task.baseline.antenna1.pos_enu)
-            task_antpos.append(task.baseline.antenna2.pos_enu)
-            task_uvw.append(task.baseline.uvw)
-
-    antnames, ant_indices = np.unique(task_antnames, return_index=True)
-    task_antnums = np.array(task_antnums)
-    task_antpos = np.array(task_antpos)
-    antnums = task_antnums[ant_indices]
-    antpos = task_antpos[ant_indices]
-
-    freqs = np.unique(task_freqs)
-
-    uv_obj = UVData()
+    history += ' Npus = ' + str(mpi.Npus) + '.'
 
-    # add pyuvdata version info
-    history += uv_obj.pyuvdata_version_str
+    uv_obj = copy.deepcopy(uv_in)
 
-    uv_obj.telescope_name = telescope_name
-    uv_obj.telescope_location = np.array([tl.to('m').value for tl in telescope_location])
-    uv_obj.instrument = telescope_name
-    uv_obj.Nfreqs = freqs.size
-    uv_obj.Ntimes = np.unique(task_times).size
-    uv_obj.Nants_data = antnames.size
-    uv_obj.Nants_telescope = uv_obj.Nants_data
-    uv_obj.Nblts = len(ant_1_array)
-
-    uv_obj.antenna_names = antnames.tolist()
-    uv_obj.antenna_numbers = antnums
-    antpos_ecef = uvutils.ECEF_from_ENU(antpos, *uv_obj.telescope_location_lat_lon_alt) - uv_obj.telescope_location
-    uv_obj.antenna_positions = antpos_ecef
-    uv_obj.ant_1_array = np.array(ant_1_array)
-    uv_obj.ant_2_array = np.array(ant_2_array)
-    uv_obj.time_array = np.array(task_times)
-    uv_obj.uvw_array = np.array(task_uvw)
-    uv_obj.baseline_array = uv_obj.antnums_to_baseline(ant_1_array, ant_2_array)
-    uv_obj.Nbls = np.unique(uv_obj.baseline_array).size
+    uv_obj.object_name = 'zenith'
```

Comment:
hard-coding this to zenith strikes me as a little odd. At a minimum I think we should allow it to be set in some way to something meaningful.

Reply:
This is currently how the object name is set on master, but I agree it should have a more meaningful name. Maybe the catalog name?

Reply:
Maybe catalog + something about the RA/Dec of zenith during the sim -- This is usually used in radio astronomy to indicate what object(s) or parts of the sky the observation was looking at. We could also allow people to set it in some way that was meaningful to them...